### ISSUE TITLE
Allow Combining looped ways

### ISSUE THREAD

**Issue by blackboxlogic** [openstreetmap/iD#7732]
In the situation where a road ends in a turning loop, mapped as separate elements (in a "lollipop" shape), I'd like to be able to select both elements, and use the 'combine' action to turn them into one element.

An example:
- Select way/11418145 and way/11416675
- Type the letter 'C' to combine them
- Receive error: "These features can't be merged because their endpoints aren't connected."

There might be a question of which end of the loop should connect to the straight line, I would pick the end which maintains the way's direction. If the straight way is "leaving" the loop, use the loop's end. If the straight way is arriving at the loop, use the loop's start.

The current process for combining elements like this seems to be: split the loop in half, combine each half with the straight part, which seems to work _sometimes_.

**Edit**
After a another look, I've found that the loop (way/11418145) doesn't have its start/end point where the straight part connects. I see that this makes the issue more technically complicated, and yet, as a user _without_ this knowledge, I know _exactly_ how I would expect it to behave (just combine them).
Perhaps there are situations, other than roads, where the start/end point of a loop is significant (maybe like a racetrack?), in which case, it might be incorrect to rotate a loop around itself just to combine it with another element?

![image](https://user-images.githubusercontent.com/31355456/85571850-7e195280-b602-11ea-8c22-0aba9801a5ce.png)

**Comment by blackboxlogic** [comment]
A [brief discussion](https://osmus.slack.com/archives/C2VJAJCS0/p1593007575251100) found no cases where this would break something.

**Comment by bhousel** [comment]
> A [brief discussion](https://osmus.slack.com/archives/C2VJAJCS0/p1593007575251100) found no cases where this would break something.

Combining the loop with the straight part could possibly break a relation with a "via" node at the intersection point, such as a turn restriction, enforcement, or speed camera or something, but this would be pretty rare.

**Comment by jidanni** [comment]
Yes, wanting to connect "O" + "----------" is very common in the land of driveways that end in flowerbed circles, when one wants to map in detail.

If all this can be done with future added magic, care could be added to assume we want the direction of the circle end to be in the fashion of what way the country drives: US vs. UK etc. Yes, even if we are not intending to later declare it as a one-way street. But still a better than "flip of a coin" choice. (Well, one could say "just draw your circle in the way you want it to run in the first place.")
